src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabMergeRequestSCMEvent.java

```diff
@@ -109,42 +114,42 @@ public String description() {
                 .newRequest(source, null)) {
             MergeRequestEvent.ObjectAttributes m = getPayload().getObjectAttributes();
             String originOwner = getPayload().getUser().getUsername();
+            // This is a hack to get the namespace from the path with namespace
             String originProjectPath = m.getSource().getPathWithNamespace();
-            Set<ChangeRequestCheckoutStrategy> strategies = request.getMRStrategies(
-                    source.getProjectPath().equalsIgnoreCase(originProjectPath)
-            );
-            for (ChangeRequestCheckoutStrategy strategy : strategies) {
-                MergeRequestSCMHead h = new MergeRequestSCMHead(
-                        "MR-" + m.getIid() + (strategies.size() > 1 ? "-" + strategy.name()
-                                .toLowerCase(Locale.ENGLISH) : ""),
-                        m.getId(),
-                        new BranchSCMHead(m.getTargetBranch()),
-                        ChangeRequestCheckoutStrategy.MERGE,
-                        originProjectPath.equalsIgnoreCase(source.getProjectPath())
-                                ? SCMHeadOrigin.DEFAULT
-                                : new SCMHeadOrigin.Fork(originProjectPath),
-                        originOwner,
-                        originProjectPath,
-                        m.getSourceBranch());
-                GitLabApi gitLabApi = GitLabHelper.apiBuilder(source.getServerName());
-                // Since rn there is no way to get the commit hash of target branch fetch the mr again
-                // A better way can be searching for the target branch head and getting the commit hash
-                MergeRequest mr = gitLabApi.getMergeRequestApi().getMergeRequest(source.getProjectPath(), m.getIid());
+            int namespaceLength = originProjectPath.lastIndexOf("/");
+            originProjectPath = originProjectPath.substring(0, namespaceLength);
+            Map<Boolean, Set<ChangeRequestCheckoutStrategy>> strategies = request.getMRStrategies();
+            boolean fork = !getPayload().getUser().getUsername().equals(originOwner);
```

Comment:
```suggestion
            boolean fork = !getPayload().getUser().getUsername().equals(originOwner);
```

This is logically just comparing the username and will be false?

Reply:
Yes this needs a similar change like #17

Reply:
Fixed in 3bd7198